src/plugins/visualizations/public/legacy/build_pipeline.ts

```diff
@@ -550,3 +577,93 @@ export const buildPipeline = async (
 
   return pipeline;
 };
+
+export async function buildVisExpression(
+  vis: Vis,
+  params: {
+    timefilter: TimefilterContract;
+    timeRange?: TimeRange;
+    abortSignal?: AbortSignal;
+  }
+) {
+  const { indexPattern, savedSearchId, searchSource } = vis.data;
+  const { uiState } = vis;
+  const query = searchSource!.getField('query');
+  const filters = searchSource!.getField('filter');
+
+  // context
+  const pipeline = buildExpression([buildExpressionFunction('kibana', {})]);
+  const kibanaContext = buildExpressionFunction('kibana_context', {});
+
+  if (query && typeof query === 'string') {
+    // TODO:
+    // The typings for `Query` are `string | { [key: string]: any }
+    // Will it ever be in object format when coming from vis.data?
+    kibanaContext.addArgument('q', query);
+  }
+
+  if (filters) {
+    kibanaContext.addArgument('filters', JSON.stringify(filters));
+  }
+
+  if (savedSearchId) {
+    kibanaContext.addArgument('savedSearchId', savedSearchId);
+  }
+
+  // request handler
+  if (vis.type.requestHandler === 'courier' && indexPattern?.id) {
+    pipeline.functions.push(
```

Comment:
just a note .... once esaggs function is refactored we'll be able to do `pipeline.functions.push(vis.data.aggs.toExpressionAST())`